pkg/controller.v1/common/job.go

```diff
@@ -38,13 +39,13 @@ import (
 	volcanov1beta1 "volcano.sh/apis/pkg/apis/scheduling/v1beta1"
 )
 
-func (jc *JobController) DeletePodsAndServices(runPolicy *apiv1.RunPolicy, job interface{}, pods []*corev1.Pod) error {
+func (jc *JobController) DeletePodsAndServices(runPolicy *apiv1.RunPolicy, runtimeObject runtime.Object, pods []*corev1.Pod) error {
 	if len(pods) == 0 {
 		return nil
 	}
 
-	// Delete nothing when the cleanPodPolicy is None.
-	if *runPolicy.CleanPodPolicy == apiv1.CleanPodPolicyNone {
+	// Delete nothing when the cleanPodPolicy is None and the runPolicy.suspend is false.
+	if !trainutil.IsJobSuspended(runPolicy) && *runPolicy.CleanPodPolicy == apiv1.CleanPodPolicyNone {
```

Comment:
What is the reasoning for this changes. Should it be the caller's check to only cleanup resources if the job is suspended (or for other reasons, such as succeeded) ? 

Also, it looks suspicious in this form, because the cleanup will be performed when the job is suspended, and the cleanup policy is `None`.

Reply:
> What is the reasoning for this changes.

I changed it here since I would clean up resources even if the cleanPodPolicy is `None`, like [MPIJob](https://github.com/kubeflow/mpi-operator/blob/18250f5e6980ce3afbf86359f8aa5fa9ac6cf831/pkg/controller/mpi_job_controller.go#L648-L653).

> Also, it looks suspicious in this form, because the cleanup will be performed when the job is suspended, and the cleanup policy is None.

Yes, the behavior is expected. 
@mimowo Does that mean we shouldn't remove resources if the cleanPodPolicy is `None`?

Reply:
> I changed it here since I would clean up resources even if the cleanPodPolicy is `None`, like [MPIJob](https://github.com/kubeflow/mpi-operator/blob/18250f5e6980ce3afbf86359f8aa5fa9ac6cf831/pkg/controller/mpi_job_controller.go#L648-L653).

I see, thanks for pointing out. 

> Yes, the behavior is expected.
> @mimowo Does that mean we shouldn't remove resources if the cleanPodPolicy is `None`?

I think it is ok to clean the resources, as MPIJob does, since the policy only declares what is happening when the job is finished: https://github.com/kubeflow/mpi-operator/blob/master/pkg/apis/kubeflow/v2beta1/types.go#L43-L44. 

Now I see, the `!IsJobSuspended` is an indirect way of saying `IsJobFinished`, or `commonutil.IsSucceeded(jobStatus) || commonutil.IsFailed(jobStatus)`, but this wasn't obvious when I was reading, and this is a little bit fragile, because it assumes there are only two mutually exclusive scenarios when the method is called.

I would suggest to either check `isJobFinished`  so that the condition matches the documentation, or expand the comment to explain that by requiring `suspend=false` we indirectly check that the job is finished.

Reply:
That makes sense.
I will add a check if the job is finished (`isJobFinished`). Thanks!

Reply:
Done.